jupyterlab 3.5.2

skupr-anaconda opened the pull request:
**Jira ticket:** [PKG-858](https://anaconda.atlassian.net/browse/PKG-858) (jupyterlab-3.5.2)

**The upstream data:**
Github releases:  https://github.com/jupyterlab/jupyterlab/releases
[Diff between the latest and previous upstream releases](https://github.com/jupyterlab/jupyterlab/compare/v3.5.0...v3.5.2)
License: https://github.com/jupyterlab/jupyterlab/blob/master/LICENSE
Changelog: https://github.com/jupyterlab/jupyterlab/blob/master/CHANGELOG.md
Requirements:
 * setup.py:  https://github.com/jupyterlab/jupyterlab/blob/v3.5.2/setup.py
 * setup.cfg: https://github.com/jupyterlab/jupyterlab/blob/v3.5.2/setup.cfg
 *  pyproject.toml:  https://github.com/jupyterlab/jupyterlab/blob/v3.5.2/pyproject.toml

**_Actions:_**

- Fix pinning of nodejs in test/requires


**Additional info:**

<details>

**Package's statistics**

<details>

 * Priority B | effort: hard | Category: anaconda | subcategory: core | pkg_type: python
 * Outdated platfroms: 'linux-64', 'linux-aarch64', 'linux-ppc64le', 'osx-64', 'osx-arm64', 'win-64'
 * Latest version: 3.5.2
 * Release on PyPi:
    * version: 3.5.2
    * date: 2022-12-19T12:21:46
* [PyPi history](https://pypi.org/project/jupyterlab/#history)
 * Popularity: 
    * 3 months downloads: 1038473.0
    * All time:  7034343 downloads -  jupyterlab  3.5.2  An extensible environment for interactive and reproducible computing, based on the Jupyter Notebook and Architecture.  

</details>

**Other checks:**

<details>

1. - [x] https://github.com/jupyterlab/jupyterlab/tree/v3.5.2 exists
2. - [ ] The upstream url error:
    https://github.com/jupyterlab/jupyterlab
    
3. - [ ] Check the pinnings
4. - [ ] Changelog url error:
    https://github.com/jupyterlab/jupyterlab/tree/v3.5.2
    200
5. - [ ] Additional research
    https://github.com/jupyterlab/jupyterlab/issues
    
6. - [ ] `dev_url` error:
    https://github.com/jupyterlab/jupyterlab
    
7. - [ ] `doc_url` error:
    https://jupyterlab.readthedocs.io
    
8. - [ ] Verify that the `build_number` is correct
9. - [x] has `setuptools`
10. - [x] has `wheel`
11. - [x] `pip` in test
12. - [ ] Verify the test section
13. - [ ] Verify if the package is `architecture specific`
14. - [ ] Verify that private modules are not mentioned in the recipe For example: (_private_module)
15.  - [ ] license_file doesn't exist!

16. - [x] license_family BSD is present
17. - [x] License: BSD-3-Clause
    - [x] License is `spdx` compliant
 * Check if the license identifier has correct name from the SPDX License List

| Identifier                           |
|:-------------------------------------|
| BSD-3-Clause                         |
| BSD-3-Clause-Attribution             |
| BSD-3-Clause-Clear                   |
| BSD-3-Clause-LBNL                    |
| BSD-3-Clause-Modification            |
| BSD-3-Clause-No-Military-License     |
| BSD-3-Clause-No-Nuclear-License      |
| BSD-3-Clause-No-Nuclear-License-2014 |
| BSD-3-Clause-No-Nuclear-Warranty     |
| BSD-3-Clause-Open-MPI                |
</details>

**Check GUI app:**

<details>
**GUI app check**

Recommend to make testing in c3i_test2 channel!
../aggregate/jupyterlab is a GUI aplication

**End GUI check**

</details>

**Check dependency issues:**

<details>
../aggregate/jupyterlab-feedstock/recipe/meta.yaml
Dependencies: ['ipython', 'tomli', 'setuptools', 'nbclassic', 'tornado >=6.1.0', 'jupyter-packaging >=0.9,<2', 'wheel', 'pip', 'jupyter_core', 'python', 'packaging', 'notebook <7', 'jinja2 >=2.1', 'jupyter_server >=1.16,<3', 'jupyterlab_server >=2.10,<3']


noarch:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-ppc64le:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

linux-s390x:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

osx-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

osx-arm64:
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

win-64:
- py3.7:  No issues found
- py3.8:  No issues found
- py3.9:  No issues found
- py3.10:  No issues found

</details>


**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/jupyterlab-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/jupyterlab-feedstock/tree/3.5.2)
* [conda-forge recipe](https://github.com/{upstream}/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/jupyterlab)
* [Diff between upstream and feature branch](https://github.com/conda-forge/jupyterlab-feedstock/compare/main...AnacondaRecipes:3.5.2)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/jupyterlab-feedstock/compare/master...AnacondaRecipes:3.5.2)
* [The last merged PRs](https://github.com/AnacondaRecipes/jupyterlab-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)

</details>

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 3.5.2 git@github.com:AnacondaRecipes/jupyterlab-feedstock.git
```

skupr-anaconda committed:
DOR update to 3.5.2

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_03ajfntv1n/clone/recipe/meta.yaml:93: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_03ajfntv1n/clone/recipe/meta.yaml:93: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

skupr-anaconda committed:
Update to 3.5.2

- Fix pinning of nodejs in test/requires

- Enable downstreams for testing

[PKG-858]

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_9a_t51y_8v/clone/recipe/meta.yaml:93: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_9a_t51y_8v/clone/recipe/meta.yaml:93: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

skupr-anaconda committed:
Disable downstreams

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_837j0ci7dn/clone/recipe/meta.yaml:93: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_837j0ci7dn/clone/recipe/meta.yaml:93: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

ryanskeith reviewed (approved):
LGTM

ELundby45 reviewed (approved):
LGTM

varlackc reviewed (approved):
LGTM

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_210h_eoojf/clone/recipe/meta.yaml:93: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_210h_eoojf/clone/recipe/meta.yaml:93: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_76ft8wix7w/clone/recipe/meta.yaml:93: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_76ft8wix7w/clone/recipe/meta.yaml:93: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>
